common/arg.cpp

```diff
@@ -1844,6 +1882,43 @@ common_params_context common_params_parser_init(common_params & params, llama_ex
             params.hf_token = value;
         }
     ).set_env("HF_TOKEN"));
+    add_opt(common_arg(
+        {"-ms", "-msr", "--ms-repo"}, "<user>/<model>[:quant]",
+        "ModelScope model repository; quant is optional, case-insensitive, default to Q4_K_M, or falls back to the first file in the repo if Q4_K_M doesn't exist.\n"
+        "example: unsloth/phi-4-GGUF:q4_k_m\n"
+        "(default: unused)",
+        [](common_params & params, const std::string & value) {
+            params.hf_repo = value;
```

Comment:
I think this is not very intuitive to read, a bit hacky here.

Cleaner way is to rename `hf_repo` to `url_repo` (same for `hf_file`) and add `url_domain` to switch between different hosts.